app/src/lib/stores/pull-request-store.ts

```diff
@@ -0,0 +1,71 @@
+import { PullRequestDatabase } from '../databases'
+import { GitHubRepository } from '../../models/github-repository'
+import { Account } from '../../models/account'
+import { API, IAPIPullRequest } from '../api'
+import { fatalError } from '../fatal-error'
+
+/** The store for GitHub Pull Requests. */
+export class PullRequestStore {
+  private db: PullRequestDatabase
+
+  public constructor(db: PullRequestDatabase) {
+    this.db = db
+  }
+
+  public async cachePullRequests(
+    repository: GitHubRepository,
+    account: Account
+  ) {
+    const api = API.fromAccount(account)
+
+    const prs = await api.fetchPullRequests(
+      repository.owner.login,
+      repository.name,
+      'open'
+    )
+
+    this.writePullRequests(prs, repository)
```

Comment:
We should `await` this, otherwise the `Promise` returned from `cachePullRequests` will resolve before this is done.